Labyrinth/src/LabyrinthOriginal/PushRowMode.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * We should be able to use the PushRowMode to implement the board size restriction.
+ * We won't let the user use the push tile mode - we might just want to get rid of
+ * it all together... And no other modes allow pushing a tile off the board.
+ */
+package LabyrinthOriginal;
+
+import java.awt.Point;
+
+import Physics.ActionMode;
+import Physics.Board;
+import Physics.Cursor;
+import Physics.Directions;
+import Physics.Tile;
+
+public class PushRowMode extends ActionMode {
+	Board board;
+	Tile extraTile;
+	
+	public PushRowMode(Board b) {
+		this.board = b;
+		this.extraTile = new Tile(-1,-1); //Arbitrary location - not on a board so doesn't matter.
+		this.extraTile.setUnblocked("Up", true);
+		this.extraTile.setUnblocked("Right", true);
+		this.extraTile.setUnblocked("Down", true);
+		this.extraTile.setUnblocked("Left", true);
```

Comment:
Can we put this into a method `board.generateRandomTile` or something? We talked about remaking the board generation method as well using something like `board.generateRandomTile(distribution)`.